docs/developer/inventory.md

```diff
@@ -0,0 +1,164 @@
+| `extract.globs`       | no  | a list of glob strings to match specific files from an archive | Any valid glob such as `*.exe`, `bin/**/asset` ([options](https://docs.python.org/3/library/fnmatch.html)). Will extract all files in the archive if omitted.  |
+| `extract.flatten`     | no  | copy the files to the destination root | `true` or `false` (default) |
+
+## Running the Script
+
+The inventory script supports three commands:
+- `lock`
+- `sync`
+- `list`
+
+The following sections assume you invoke the script with `poetry run`.
+
+#### lock
+
+The `lock` command updates the lock file based on the configuration defined in
+`pyproject.toml` / `inventory.toml`. For each asset, it reads its details,
+queries GitHub to find the appropriate release and asset URL, computes a
+checksum if the asset is available locally, uses caching for fetching and
+hashing, renders filenames that contain the `{version}` template string, and
+supports assets that are platform-agnostic using `platform.all`.
+
+Example:
+
+```
+./dev_scripts/inventory.py lock
+Processing 'asset1'
+Processing 'asset2'
+Lock file 'inventory.lock' updated.
+```
+
+#### sync
+
+The `sync` command synchronizes (downloads or copies) assets as specified in the
+lock file for the given platform (or the current platform if none is provided).
+It downloads assets into a cache, verifies them against an expected hash, copies
+them to the destination, marks files as executable if required, and extracts
+files based on the provided extraction criteria.
+
+Examples:
+
+Sync all assets for the current platform:
+
+```
+./dev_scripts/inventory.py sync
+Syncing 'asset1'
+Syncing 'asset2'
+Synced 2 assets.
+```
+
+Sync all assets for the provided platform:
+
+```
+./dev_scripts/inventory.py sync -p darwin/amd64
+Syncing 'asset3'
+Synced 1 assets.
+```
+
+Sync only specific assets:
+
+```
+./dev_scripts/inventory.py sync asset1
+Syncing 'asset1'
+Synced 1 assets.
+```
+
+#### list
+
+The `list` command lists all assets defined for a specific platform, or the
+current one, if not specified. The list output contains the name of the asset,
+its version, and its download URL.
+
+Example:
+
+```
+./dev_scripts/inventory.py list
+asset1 0.0.1 https://github.com/owner/repo/releases/download/v0.0.1/asset1
+asset2 1.2.3 https://github.com/owner/other/releases/download/v0.0.1/asset2
+```
+
+Pass `-vv` to get full details for each asset entry.
+
+### Common Arguments
+
+Each command supports the following optional arguments:
+
+- `-p, --platform`:
+  Specify the platform for which the assets should be processed. Examples
+  include: `windows/amd64`, `linux/amd64`, `darwin/amd64`, `darwin/arm64`.
+  If not provided, the current platform is auto-detected.
+
+- `-v, --verbose`:
+  Enable verbose logging. Use `-v` for INFO level or `-vv` (or more) for DEBUG
+  level messaging.
+
+- `-C, --directory`:
```

Comment:
I'm curious: why is this `-C` and not `-d`? Why in caps?

Reply:
Blatantly copy-pasted from Poetry, who have copy-pasted it from some other Unix tools :grimacing: If you feel too strong about it, I can change it though.

Reply:
Yeah, yeah, let's go with `-d`.

Reply:
I don't see where the `directory` option is used at all. Maybe you could point me to it? :thinking:

Reply:
Damn, it totally slipped! Check out b0ca4c4b6b8d4ed5ffb2227d7220b198ade01b60 for a fix.